### PR TITLE
Enhacement line plots

### LINKED ISSUE #48: Spectrogram has wrong FFT normalization
## Description

pyfar.dsp.spectrogram has the wrong FFT normalization. A spectrogram of a sine ('power' signal with 'amplitude' normalization) should should show maximum values of 0 dB but it does not. This is caused by the scipy.signal.spectrogram normalization, which is less flexible than the pyfar.fft.normalization.

### PR DESCRIPTION
**Pull request by f-brinkmann**
### Which issue(s) are closed by this pull request?

Closes #48, related to #46. Merge after #54. Next step: refactoring interaction

### Changes proposed in this pull request:

**`pyfar.plot.line`**
- merged `time()` and `time_dB()` plots by adding a `dB`-option. The default is `dB=False`
- added a `dB`-option to the `freq()` plot. The default is `dB=True`
- added an `xscale`-option to all plots that have a frequency-axis. The default is `xscale='log'`

**`pyfar.plot.line.spectrogram`**
- added auto range for colormap if `dB=True` for consistency with pyfar.plot.line.freq
- added colorbar clipping

**`test_plot.py`**
- added a MagicMock for Signal that can handle `__getitem__`

**`pyfar.dsp.spectrogram`**
- removed clipping the spectrogram data. Clipping is now done by clipping the colormap inside the spectrogram plots.
- added correct normalization using pyfar.fft.normalization
- removed `window_length='auto'` in favor of `window_length=1024`. The 'auto'-option changed the `window_length` depending on the signal length, which 
- added `dB`, `log_prefix` and `log_reference` as input parameters (`dB`replaces old input `nodb` and is used for consistency with #45)
- time stamps now refer to the beginning of the blocks, i.e., they start at 0 seconds. Before, they referred to the center of the block, which looks less nice in plots. Both conventions are used regularly, the current one might be more common.

### CODE REVIEW

File: pyfar/dsp/dsp.py [hunk 1]
```diff
@@ -223,3 +119,77 @@ def nextpow2(x):
...
+    # apply normalization from signal
+    spectrogram = fft.normalization(
+        spectrogram, window_length, signal.sampling_rate,
+        signal.fft_norm, window=window)
+
+    # get in dB
+    if dB:
+        eps = np.finfo(float).tiny
+        spectrogram = log_prefix*np.log10(
+            np.abs(spectrogram) / log_reference + eps)
```
Comment (mberz) [inline pyfar/dsp/dsp.py:189]: I think I would not implement the dB option here, but in the respective plot function.

File: pyfar/plot/_line.py [hunk 2]
```diff
@@ -204,32 +182,44 @@ def _freq(signal, log_prefix=20, log_reference=1, ax=None, **kwargs):
 
     # prepare input
     kwargs = _return_default_colors_rgb(**kwargs)
-    eps = np.finfo(float).tiny
-    data_dB = log_prefix*np.log10(np.abs(signal.freq)/log_reference + eps)
-    ymax = np.nanmax(data_dB)
-    ymin = ymax - 90
-    ymax = ymax + 10
+    if dB:
+        eps = np.finfo(float).tiny
```
Comment (mberz) [inline pyfar/plot/_line.py:186]:
```suggestion
        eps = np.finfo(float).eps
```

File: pyfar/plot/_line.py [hunk 3]
```diff
@@ -169,31 +143,35 @@ def _time_dB(signal, log_prefix=20, log_reference=1, ax=None, **kwargs):
...
-    # avoid any zero-values because they result in -inf in dB data
-    eps = np.finfo(float).tiny
-    data_dB = log_prefix * np.log10(np.abs(y_data) / log_reference + eps)
-    ymax = np.nanmax(data_dB)
-    ymin = ymax - 90
-    ymax = ymax + 10
+    data = signal.time.T
+    if dB:
+        # avoid any zero-values because they result in -inf in dB data
+        eps = np.finfo(float).tiny
```
Comment (mberz) [inline pyfar/plot/_line.py:149]:
```suggestion
        eps = np.finfo(float).eps```

**Comment by mberz** [comment]
One general comment: The spectrogram is not really a line plot, but currently in the line plot submodule. Doesn't really matter that much, but might lead to confusion. Maybe we can move this some place else at a later time.

**Review by sikersten — APPROVED** [review]
In many of the plot functions, the parameter is called `db` in the docstrings while it's `dB` in the function call. Beyond that, I have nothing to criticize :)

**Pull request merged**